examples/adhoc/README.md

```diff
@@ -1,55 +1,53 @@
 
-Note: The example programs are toy examples but they share some interesting properties: their running time should take around 1 minute, and their profiling data should change slightly from run to run (making them good examples for comparison / diff mode visualizations).
+### Option 1: Push mode (with SDK installed)
+If your application already uses an agent or has integration with the Pyroscope HTTP API, you can use push mode 
+to profile the application without any additional configuration changes.
 
-### Push Mode
+#### Golang Adhoc (Push Mode)
 If the application to profile is already using an agent or has some integration with the HTTP API,
 push mode can be used to profile the application without any configuration change.
 
-#### Golang adhoc (push)
 ```
 # no spy is supported, --push is not needed (but can still be provided):
 pyroscope adhoc go run adhoc-push.go
 ```
+
 #### Python adhoc (using pip package)
 ```
 # pyspy is autodetected, --push is mandatory.
 # Note that you need pyroscope-io >= 0.6.0 for this to work.
 pyroscope adhoc --push python adhoc-push.py
 ```
 
+### Option 2: Push Mode (no SDK installed -- Pyroscope sidecar)
+
+If you don't have the Pyroscope SDK installed and want to profile a Python or Ruby application, you can still push data
+to the Pyroscope server using the adhoc mode as a sidecar.
 
-### Exec mode
+#### Python adhoc (no pip package -- Pyroscope sidecar)
 For languages with a supported spy and no other pyroscope integration, this is the easiest way to get profiling data.
 For example, this method will work for python or ruby _without_ the pip/gem instrumented in the code.
 
+Profile a script using adhoc
 ```
 # Run with spy-name autodetected.
-pyroscope adhoc python adhoc-spy.py
-
-# Alternatively, specify the spy-name if it cannot be autodetected.
-pyroscope adhoc --spy-name pyspy ./adhoc-spy.py
+./bin/pyroscope adhoc --push   ~/Downloads/pyroscope-cli exec --spy-name=pyspy   examples/python/simple/main.py  
 ```
 
-### Connect mode
-#### Profile a process that is already running
-
-If the profiled process is already running, it's possible to attach to it instead, indicating its PID through the `--pid` flag:
+Attach to and profile a process using adhoc
+```
+sudo ./pyroscope-cli connect --pid=[369936] --spy-name=pyspy --server-address="http://localhost:4100"
+```
 
+Using docker:
 ```
-# Run the program normally, in the background. It should give the PID as output.
-python adhoc-spy.py &
-# => [1] 841690
-# Use adhoc to attach to the running command.
-# Note that the --spy-name is now mandatory as it cannot be inferred.
-# Also, pyroscope needs to be launched as root to be able to trace the running process.
-sudo pyroscope adhoc --spy-name pyspy --pid 841690
+ docker run --rm -ti --privileged --pid=host pyroscope/pyroscope-rs-cli:0.2.7-457bb15 connect --pid=369936 --spy-name=pyspy --server-address="http://localhost:4100"
```

Comment:
```suggestion
```
This is not really adhoc, just connecting to a running process without installing libraries, this should go to a separate docs page, describing how to use pyroscope-cli from pyroscope-rs